### PR TITLE
Adds CircleCI configuration

### PR DESCRIPTION
**Pull request by tinynumbers**
## OVERVIEW
This is an experiment to explore the utilization of CircleCI for continuous integration testing of this repo.

## WHERE SHOULD THE REVIEWER START?
`.circleci/config.yml`

## HOW CAN THIS BE MANUALLY TESTED?
N/A

## ANY NEW DEPENDENCIES ADDED?
No.

## CHECKLIST
_Be sure all items are_ ✅ _before submitting a PR for review._
* [x] Verify the linter and tests pass: `npm run review`
* [x] Verify this branch is rebased with the latest master

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Javascript Node CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-javascript/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version you desire here
+      - image: circleci/node:6.10.3
```
Comment (andrewtpoe) [inline .circleci/config.yml]: Not LTS?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          - v1-dependencies-
+
+      - run: yarn install
+
+      - save_cache:
+          paths:
+            - node_modules
+          key: v1-dependencies-{{ checksum "package.json" }}
+        
+      # run tests!
```
Comment (andrewtpoe) [inline .circleci/config.yml]: Should we also include a run lint section?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# Check https://circleci.com/docs/2.0/language-javascript/ for more details
+#
+version: 2
+jobs:
+  build:
+    docker:
+      # specify the version you desire here
+      - image: circleci/node:6.10.3
+      
+      # Specify service dependencies here if necessary
```
Comment (andrewtpoe) [inline .circleci/config.yml]: Do we need to keep this block?

File: .circleci/config.yml (hunk 2 above)
Comment (tinynumbers) [reply .circleci/config.yml]: Actually yeah it should probably just be `yarn review` not `yarn test`.

File: .circleci/config.yml (hunk 1 above)
Comment (tinynumbers) [reply .circleci/config.yml]: Take your pick:
https://circleci.com/docs/2.0/circleci-images/#nodejs

File: .circleci/config.yml (hunk 3 above)
Comment (tinynumbers) [reply .circleci/config.yml]: Maybe for documentation purposes?

File: .circleci/config.yml (hunk 1 above)
Comment (alanbsmith) [reply .circleci/config.yml]: `- image: circleci/node:latest`?

Comment (andrewtpoe) [reply .circleci/config.yml]: Yeah, I think I like `latest` as well. Kind of surprised they don't have an `LTS` option to track what the current lts is.

**Pull request merged**